src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -102,6 +102,21 @@
     /** Maximum time we allow for elapsed fsync before WARNing */
     private final static long fsyncWarningThresholdMS;
 
+    /**
+     * This parameter limit the size of each txnlog to a given limit (KB).
+     * It does not affect how often the system will take a snapshot [zookeeper.snapCount]
+     * We roll the txnlog when either of the two limits are reached.
+     * Also since we only roll the logs at transaction boundaries, actual file size can exceed
+     * this limit by the maximum size of a serialized transaction.
+     * The feature is disabled by default (-1)
+     */
+    public static final String LOG_SIZE_LIMIT = "zookeeper.txnlogSizeLimitInKb";
```

Comment:
shouldn't it be txnLogSizeLimitInKb?

Reply:
we should also add this property to the doc.

Reply:
Thanks @breed, are you referring to https://github.com/apache/zookeeper/blob/master/docs/zookeeperAdmin.html?

Reply:
Thanks @anmolnar I will update the patch soon with documentation.